### PR TITLE
test(storage): add bucket conditions test

### PR DESCRIPTION
**Pull request by tritone**
Add an emulator test for bucket conditions (similar to the object one added previously). This covers all bucket methods which take preconditions and did not catch any bugs.

Also minor formatting cleanup to the object test.

### CODE REVIEW

File: storage/client_test.go [hunk 1]
```diff
@@ -1134,7 +1194,7 @@ func TestObjectConditionsEmulated(t *testing.T) {
 			call func() error
 		}{
 			{
-				name: "Object update generation",
```
Comment (BrennaEpp) [inline storage/client_test.go:1137]: Nice

File: storage/client_test.go [hunk 2]
```diff
@@ -1120,6 +1120,66 @@ func TestHMACKeyCRUDEmulated(t *testing.T) {
...
+			name string
+			call func() error
+		}{
+			{
+				name: "get",
+				call: func() error {
+					bucket, metaGen, err := createBucket(ctx, project)
+					if err != nil {
+						return fmt.Errorf("creating bucket: %w", err)
+					}
```
Comment (BrennaEpp) [inline storage/client_test.go:1136]: Can this be refactored out into t.Run and passed into the call?

**Review by BrennaEpp — APPROVED** [review]
LGTM, just a comment on minor refactoring

File: storage/client_test.go (hunk 2 above)
Comment (tritone) [reply storage/client_test.go:1136]: sure, done

**Pull request merged**